src/main/java/com/frcteam3255/components/motors/SN_SparkMax.java

```diff
@@ -0,0 +1,194 @@
+        }
+        else {
+            throw new java.lang.UnsupportedOperationException("Only DemandType.ArbitraryFeedForward is supported.");
+        } 
+    }
+
+    @Override
+    public void setNeutralMode(NeutralMode neutralMode) {
+        switch (neutralMode) {
+            case Brake:
+                super.setIdleMode(IdleMode.kBrake);
+                break;
+            case Coast:
+                super.setIdleMode(IdleMode.kCoast);
+                break;
+        
+            default:
+                break;
+        }
+    }
+
+    
+    @Override
+    public ErrorCode configAllSettings(BaseTalonConfiguration allConfigs) {
+        // Set slot0 PID
+        pidController.setP(allConfigs.slot0.kP,0);
+        pidController.setD(allConfigs.slot0.kD,0);
+        pidController.setI(allConfigs.slot0.kI,0);
+        pidController.setFF(allConfigs.slot0.kF,0);
+        pidController.setIZone(allConfigs.slot0.integralZone, 0);
+        pidController.setIMaxAccum(allConfigs.slot0.maxIntegralAccumulator, 0);
+        pidController.setOutputRange(-allConfigs.slot0.closedLoopPeakOutput, allConfigs.slot0.closedLoopPeakOutput, 0);
+        pidController.setSmartMotionAllowedClosedLoopError(allConfigs.slot0.allowableClosedloopError, 0);
+        
+        // Set slot1 PID
+        pidController.setP(allConfigs.slot1.kP,1);
+        pidController.setD(allConfigs.slot1.kD,1);
+        pidController.setI(allConfigs.slot1.kI,1);
+        pidController.setFF(allConfigs.slot1.kF,1);
+        pidController.setIZone(allConfigs.slot1.integralZone, 1);
+        pidController.setIMaxAccum(allConfigs.slot1.maxIntegralAccumulator, 1);
+        pidController.setOutputRange(-allConfigs.slot1.closedLoopPeakOutput, allConfigs.slot1.closedLoopPeakOutput, 1);
+        pidController.setSmartMotionAllowedClosedLoopError(allConfigs.slot1.allowableClosedloopError, 1);
+        
+        // Set slot3 PID
+        pidController.setP(allConfigs.slot3.kP,3);
+        pidController.setD(allConfigs.slot3.kD,3);
+        pidController.setI(allConfigs.slot3.kI,3);
+        pidController.setFF(allConfigs.slot3.kF,3);
+        pidController.setIZone(allConfigs.slot3.integralZone, 3);
+        pidController.setIMaxAccum(allConfigs.slot3.maxIntegralAccumulator, 3);
+        pidController.setOutputRange(-allConfigs.slot3.closedLoopPeakOutput, allConfigs.slot3.closedLoopPeakOutput, 3);
+        pidController.setSmartMotionAllowedClosedLoopError(allConfigs.slot3.allowableClosedloopError, 3);
+        
+        // Config softlimits
+        super.enableSoftLimit(SoftLimitDirection.kForward, allConfigs.forwardSoftLimitEnable);
+        super.enableSoftLimit(SoftLimitDirection.kReverse, allConfigs.reverseSoftLimitEnable);
+        super.setSoftLimit(SoftLimitDirection.kForward, (float)allConfigs.forwardSoftLimitThreshold);
+        super.setSoftLimit(SoftLimitDirection.kReverse, (float)allConfigs.reverseSoftLimitThreshold);
+
+        super.burnFlash();
+        return null;
```

Comment:
this seems bad, but doing it right would be really hard so whatever. we should fix eventually though